### ISSUE TITLE
Properly escape image URLs on upload

### ISSUE THREAD

**Issue by Throne3d** [Marri/glowfic#140]
With [this image](https://d1anwqy6ci9o1i.cloudfront.net/users/5/icons/4e2a38a5-7f57-475a-96b8-d3d47bd08cf7_kato%2029%2B.png), the `%2B` to the end of the URL had not been escaped (from `+`) by the server, and so the icon failed to display properly on the site. This should be fixed.

**Comment by Throne3d** [comment]
According to [the S3 docs for POST object](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html#RESTObjectPOST-requests-responses-response-elements), there should be a response of `Location` as well as `Key`, which (being a URI instead of the key name), may be properly escaped and fully functional as a URI.

I just uploaded a file to the site, put a breakpoint in the JavaScript, and found the following response:
```
Location: https://glowfic-constellation.s3.amazonaws.com/users%2F31%2Ficons%2F43f7f1d3-6933-4473-8221-e3f57d067853_dupe-avatar.jpg
Bucket: glowfic-constellation
Key: users/31/icons/43f7f1d3-6933-4473-8221-e3f57d067853_dupe-avatar.jpg
Etag: "722c62255530edb560f2b608c0752c8b"
```

Looking at the "Location", as given, [it does appear to be functional](https://glowfic-constellation.s3.amazonaws.com/users%2F31%2Ficons%2F43f7f1d3-6933-4473-8221-e3f57d067853_dupe-avatar.jpg) – despite the slashes and all the other things being escaped – so I think  it would be good to use that, especially as it seems to be pre-escaped for us. If the current system is used so Cloudfront can cache it, simply use this response and replace as appropriate.

The same URL is functional with the host replaced with the Cloudfront equivalent: [see here](https://d1anwqy6ci9o1i.cloudfront.net/users%2F31%2Ficons%2F43f7f1d3-6933-4473-8221-e3f57d067853_dupe-avatar.jpg).

**Comment by Throne3d** [comment]
I'm attempting some broken characters and will list if they worked below. These are listed [on a page about S3 object metadata](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html) as being problematic or potentially problematic. They may well not be escaped, in the tests prior to the relevant PR, but whether they work or break the image is listed.

The filename is `100 hilarity*broken.png`, with the `*` replaced by the relevant special character.

- `\` does not upload
- `?` uploads but does not display
- `+` uploads but does not display
- `=` works
- `  ` works (double space)
- `,` works
- `;` works
- `:` works
- `@` works
- `$` works
- `&` works

**Comment by Throne3d** [comment]
Fixed as a result of #409.
